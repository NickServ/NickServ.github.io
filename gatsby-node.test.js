const { sourceNodes } = require('./gatsby-node')

jest.mock('@octokit/rest', () => () => ({
  authenticate: () => {},
  repos: {
    get: ({ owner, repo }) =>
      Promise.resolve({
        data: {
          custom: true,
          id: require('crypto')
            .createHash('md5')
            .update(repo.toString())
            .digest('hex')
        }
      })
  }
}))

jest.mock('js-yaml', () => ({
  safeLoad: () => ({
    'Category One': [{ github: 'project-one' }],
    'Category Two': [{ github: 'project-two' }],
    Talks: [{ name: 'Talk' }]
  })
}))

test('sourceNodes', async () => {
  const createNode = jest.fn()
  await sourceNodes({ boundActionCreators: { createNode } })
  expect(createNode.mock.calls).toMatchSnapshot()
})
