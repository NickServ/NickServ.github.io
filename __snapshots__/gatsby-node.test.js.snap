// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sourceNodes 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "custom": true,
      "id": "bdca72cd7dcb0af5ec550efd7057bf13",
      "internal": Object {
        "contentDigest": "1441a7909c087dbbe7ce59881b9df8b9",
        "type": "Repository",
      },
      "parent": null,
    },
  ],
  Array [
    Object {
      "children": Array [],
      "custom": true,
      "id": "791797da1ca20829a8f7b2aedc3179dd",
      "internal": Object {
        "contentDigest": "1441a7909c087dbbe7ce59881b9df8b9",
        "type": "Repository",
      },
      "parent": null,
    },
  ],
]
`;
